docs/adr/0005-multitenancy-approach-for-marain.md

```diff
@@ -0,0 +1,263 @@
+# Multitenancy approach for Marain
+
+## Status
+
+Proposed
+
+## Context
+
+Tenancy has been made a first class citizen of all Marain services, however this by itself is not enough to make the system truly multitenanted. In order to do this, we need to determine how tenants should be created, managed and used within the Marain "world".
+
+We would like the option of deploying Marain as either a managed service, hosted by us and licenced to users as a PaaS offering, or for clients to deploy private instances into their own cloud subscriptions. We also want to give clients of the managed services the option for data to be stored in their own storage accounts or databases, but still have us run the compute aspects of the platform on their behalf.
```

Comment:
There's one more requirement that sounds nearly identical but is important in a different way: we want it to be possible for our customers to isolate their own customers' data. (So if one of our customers has their own multi-tenanted offering, we want it to be possible for them to use Marain in such a way that each of the customer's tenants gets its own storage.)